apps/torus-bridge/src/app/_components/transfer-evm.tsx

```
@@ -25,18 +25,45 @@ import { ArrowLeftRight } from "lucide-react";
 import { useMultiProvider } from "~/hooks/use-multi-provider";
 import { updateSearchParams } from "~/utils/query-params";
 
-const DEFAULT_MODE = "bridge";
-
 export function TransferEVM() {
+  const searchParams = useSearchParams();
+  const currentMode = useMemo(() => {
+    return searchParams.get("mode") as "bridge" | "withdraw" | null;
+  }, [searchParams]);
+
+  const fromChain = currentMode === "bridge" ? "Torus" : "Torus EVM";
```

Biome (1.9.4)

[error] This variable is used before its declaration.

The variable is declared here:


(lint/correctness/noInvalidUseBeforeDeclaration)